enhancements/microshift/microshift-pluggable-core-components.md

```diff
@@ -0,0 +1,213 @@
+
+* Convert `microshift` into a virtual package that depends on `microshift-core`,
+  `microshift-networking-ovn`, `microshift-storage-topolvm`, `microshift-openshift-router`
+  to preserve the current behavior.
+
+* Convert the `microshift-networking` package would be made a virtual package which
+  depends on `microshift-networking-ovn` to preserver current behavior.
+
+* The `microshift-networking-ovn` package would include the necessary
+  `/usr/lib/microshift/components.d/networking-ovn/` yamls, including crds.
+
+* The `microshift-storage-topolvm` package would include the necessary
+  `/usr/lib/microshift/components.d/storage-topolvm` yamls, including crds.
+
+* The `microshift-openshift-router` package would include the necessary
+  `/usr/lib/microshift/components.d/openshift-router` yamls, including crds.
+
+The rebase script should be updated to extract those core components
+as kustomize yamls instead of embedded assets.
+
+### Workflow Description
+
+#### Deploying
+
+The workflow for deployment is already described in the user histories.
+
+#### Upgrading
+
+Upgrading optional components would behave in the same way as upgrading
+components where the assets are embedded into the MicroShift binary.
+
+#### Configuring
+
+Configuration of the optional components is done by package selection
+when the images are created.
+
+#### Deploying Applications
+
+Applications are deployed as usual, we would add support for multiple
+directories in the user component manifests as described in the 2nd goal.
+
+### API Extensions
+
+None.
+
+### Risks and Mitigations
+
+The service manager complexity is increased, the risk of new bugs would
+be addressed by additional unit testing.
+
+The component combinations would need additional testing handled by
+an increased testing matrix.
+
+Considering networking/storage/router to be independent elements, probably
+an additional set of tests which cover "microshift-core" alone would
+be enough.
+
+### Drawbacks
+
+None.
+
+## Design Details
+
+### Open Questions [optional]
+
+1. 
+### Test Plan
+
+We will run additional test matrix with `microshift-core` alone, in those
+tests any OVN/NetworkPolicy tests would need to be disabled.
+
+Additional unit tests should cover the loading of core components as
+kustomize manifests, as well as the depencency handing between them.
+
+Additional unit tests should cover the loading of multiple `.d` style
+directories with manifests.
+
+## Implementation History
+
+* [openshift/microshift](https://github.com/openshift/microshift)
+* [Design guidelines](https://github.com/openshift/microshift/blob/main/docs/design.md)
+
+## Alternatives
+
+### controlling component behaviour from configuration/go code
+
+Instead of extracting the assets into kustomize/yaml files, we could add flags
+to the MicroShift configuration which then can control the component loading.
+
+This would complicate the handling of CRD installs, and would make development of new
```

Comment:
Lol, I don't know what was I thinking about the CRDs to be honest. It would make adding other CNIs/CSIs harder (you need to do go code, but cannot do just yaml), but other than that...